### ISSUE TITLE
limited use of the `text_epigraphy`.`type` of "stamp"

### ISSUE THREAD

**Issue by Gertrudius** [oaregithub/oare_sql#26]
There are only 4 instances of `text_epigraphy`.`type` = "stamp" and it is unclear to me whether these are erroneous entries, or whether we should consistently be rendering stamp seals as "stamp" or subsuming them all under "region" along with cylinder sealings.

SELECT * FROM text_epigraphy WHERE type = "stamp";

EDIT:
Either method of rendering it should be fine for us on the SQL side (although it might affect the front end), but the difference can impact my review of the iterative stored procedure logic.

**Comment by Gertrudius** [comment]
I spotted a small complicating factor. It appears that "stamp" occurs as the child of a line, whereas regions do not. Given that there are more than 4 stamp seals in the database, that means we are rendering some as children of lines, while the rest must be children of columns.

SELECT DISTINCT type FROM text_epigraphy WHERE parent_uuid IN (SELECT uuid FROM text_epigraphy WHERE type = "line")

**Comment by edstratford** [comment]
Stamp seals do -- usually -- occur within lines - though not always. There are cases where they form their own region (such as when on an edge of an envelope. So both possibilities occur. But there is a problem that some stamp seals are not marked accordingly. -- This will ulitmately have to be a manual process.

**Comment by Gertrudius** [comment]
Further research on this shows only 2 of the 4 instances of "stamp" have lines as their parent, while no regions at all, whether they are set as isStampSealImpression in text_markup or not, have a line as their parent.
```
SELECT * FROM text_epigraphy WHERE `type` = "stamp" AND parent_uuid IN (SELECT uuid FROM text_epigraphy WHERE type = "line");
SELECT * FROM text_epigraphy WHERE `type` = "region" AND parent_uuid IN (SELECT uuid FROM text_epigraphy WHERE type = "line");
```
At some point, we must have decided to render both cylinder and stamp seals as regions because that's the only explanation I can think of for having only 4 of the "stamp" type. Moreover, as part of that or maybe something to do with adding column objects, we must have changed the parent of virtually all the stamp seals to column. This has created an issue for my reordering scripts/procedures, as at least some retain a correct char_on_line value, but their object_on_tablet is ordered in respect to their current siblings which are mostly lines. An example of this can be seen in one of the two texts that have a type of "stamp" in them, and scroll down to object_on_tablet 215.
```
SELECT * FROM text_epigraphy WHERE text_uuid = "0834251e-cf6d-41d9-ab26-349dd08cf5ea";
```
I confirmed from a pdf of ICK 1 19a that the stamp seals are within line 19 (18 as well if we want to be nitpicky) and so their object_on_tablet placement is significantly off.

I think this means we'll have to do some major cleanup to ensure that these stamp seals are either placed back into a line or are sorted directly above or below the line they would have belonged to.

**Comment by Gertrudius** [comment]
Based on the last of the above queries, I put together this join to show which epigraphic_units have a char_on_line but do not have a type of isSealImpression in text_markup, which is a complicating issue.
```
SELECT te.uuid,te.text_uuid,te.parent_uuid,te.type,tm.type FROM text_epigraphy AS te
LEFT JOIN text_markup AS tm
	ON te.uuid = tm.reference_uuid
WHERE te.`type` = "region" 
AND te.char_on_line IS NOT NULL 
AND te.uuid NOT IN (SELECT reference_uuid FROM text_markup WHERE type = "isSealImpression");
```

**Comment by Gertrudius** [comment]
I finished going through that list of stamp seals and confirming their position in relation to surrounding lines of text wherever an image/handcopy etc. was available. The data has been updated to reflect this, and all nomenclature has been updated to region/isSealImpression in text_epigraphy and text_markup, and all regions that are seal impressions except those five which occur within a line have had their char_on_line and line values set to NULL.
